pkg/image/registry/imagesecret/rest.go

```diff
@@ -0,0 +1,53 @@
+package imagesecret
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// REST implements the RESTStorage interface for ImageStreamImport
+type REST struct {
+	secrets client.SecretsNamespacer
+}
+
+// NewREST returns a new REST.
+func NewREST(secrets client.SecretsNamespacer) *REST {
+	return &REST{secrets: secrets}
+}
+
+func (r *REST) New() runtime.Object {
+	return &kapi.SecretList{}
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &kapi.ListOptions{}, false, ""
+}
+
+// Get retrieves all pull type secrets in the current namespace. Name is currently ignored and
+// reserved for future use.
+func (r *REST) Get(ctx kapi.Context, _ string, options runtime.Object) (runtime.Object, error) {
+	listOptions, ok := options.(*kapi.ListOptions)
+	if !ok {
+		return nil, fmt.Errorf("unexpected options: %v", listOptions)
+	}
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		ns = kapi.NamespaceAll
+	}
+	secrets, err := r.secrets.Secrets(ns).List(listOptions.LabelSelector, listOptions.FieldSelector)
+	if err != nil {
+		return nil, err
+	}
+	filtered := make([]kapi.Secret, 0, len(secrets.Items))
+	for i := range secrets.Items {
```

Comment:
This looks very wrong.  This looks like a subresource on an imagestream that is an alias for something completely unrelated to that particular image stream.